pkg/instancegroups/rollingupdate.go

```diff
@@ -163,14 +187,88 @@ func (c *RollingUpdateCluster) RollingUpdate(groups map[string]*CloudInstanceGro
 		}()
 
 		wg.Wait()
-	}
 
-	for _, err := range results {
-		if err != nil {
-			return err
+		for _, err := range results {
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	glog.Infof("Rolling update completed!")
 	return nil
 }
+
+// RollingUpdateNodesPreCreate create all new nodes instance group(s) then cordons all nodes.
+// Old nodes are then drained and the old instance group(s) is deleted.
+func (r *RollingUpdateCluster) RollingUpdateNodesPreCreate(nodeGroups map[string]*CloudInstanceGroup) error {
```

Comment:
I left this in rolling-update, because it is not an instance group action, but an action on all instance groups.  Thoughts?